ERCS/erc-7715.md

```diff
@@ -46,6 +46,34 @@ The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL
 
 We introduce a `wallet_grantPermissions` method for the DApp to request the Wallet to grant permissions.
 
+#### Permissions & Policies
+
+Before we specify the JSON-RPC request and response it's important to clarify the relationship between permissions and policies.
+
+Permissions represent a singular piece of authority that can be used by the application.
+eg. ability to transfer a token, update an ENS domain, vote on a DAO, etc
+
+Policies reduce the permitted scope on any given permission. A policy without a permission could be excessively permissive.
+eg. a token allowance, a gas limit, an access list, an ENS record, etc.
+
+For each permission there will be a set of policies and for each policy there will be a set of properties which are specified as follows:
+
+```typescript
+interface Permission {
+  type: name;
+  policies: Policy[];
+  required: boolean;
+  properties: any;
+  data: any;
+}
```

Comment:
Is there much benefit to having the `Permission` with `data` arguments in addition to the `policies` list? The schema is identical, and by including it then Permission both grants _and restricts_ permission. 

Rather than
```
const Permission = {
  type: {
    name: "erc20-token-transfer",
    description: "Transfer ${ticker} tokens",
  },
  properties: {
    address: "address",
    ticker: "string",
  },
  data: {
    address: "0x...", // erc20 contract
    ticker: "USDC",
  },
  policies: []
}
```

you have

```
const Permission = {
  type: {
    name: "erc20-token-transfer",
    description: "Transfer ${ticker} tokens",
  },
  policies: [
    {
      type: {
         name: "contract-address",
         description: "Restricted to interactions with this contract",
      },
      properties: {
        address: "address",
        ticker: "string",
      },
      data: {
        address: "0x...", // erc20 contract
        ticker: "USDC",
      },
    },
  ],
};

```

This puts a more strict separation:
- Permission is granting ability to perform an Action. A Permission with no restricting policies grants full use of that Action (e.g. transfer ERC20, _any_ erc20). 
- Policies restrict the action, which may be the Spending Limit, the Destination Address, the Token, or anything else.

Reply:
But if you want to specify multiple tokens then you just specify the same permission with different data values

Otherwise in your proposed approach you can't associate a ERC20 contract address with an allowance or gas limit

Reply:
Yeah you can specify the same permission multiple times...but that's unnecessarily verbose, and still can't accomplish an "any" token permission.

You can still add any other policies to the array as before -- really the only change here is: Contract Address restriction is a Policy rule not a Permission property, just like a gas-limit or destination restriction.

```
const Permission = {
  type: {
    name: "erc20-token-transfer",
    description: "Transfer ${ticker} tokens",
  },
  policies: [
    {
      type: "contract-address",
      data: {
        address: ["0x..."], // erc20 contract
      },
    },
    {
      type: "spending-limit",
      data: {
        allowance: "0x...",
      },
    },
    {
      type: "gas-limit",
      data: {
        limit: "0x...",
        scope: "request", // per-request gas limit
      },
    },
  ],
};
```

I'm suggesting no priority is given to any specific match criteria: we don't assume ERC20 transfer always means the ability to transfer **a single token**; an equally valid scenario is the ability to transfer "any" ERC20 token to a single destination address (e.g., I give a dapp permission to sweep any tokens I get into my new account...like a forwarding address).